File: PhotographerApp.API/Controllers/Content/ContactController.cs
using FluentValidation;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using PhotographerApp.Core.DTOs;
using PhotographerApp.Core.Interfaces;

namespace PhotographerApp.API.Controllers.Content;

[ApiController]
[Route("api/[controller]")]
public class ContactController : ControllerBase
{
    private readonly IContactService _contactService;
    private readonly IValidator<SubmitContactMessageRequest> _validator;

    public ContactController(IContactService contactService, IValidator<SubmitContactMessageRequest> validator)
    {
        _contactService = contactService;
        _validator = validator;
    }

    [AllowAnonymous]
    [HttpPost]
    public async Task<ActionResult<ApiResponse>> Submit(
        [FromBody] SubmitContactMessageRequest request,
        CancellationToken cancellationToken)
    {
        var validation = await _validator.ValidateAsync(request, cancellationToken);
        if (!validation.IsValid)
        {
            return BadRequest(new ApiResponse
            {
                Success = false,
                Message = "Invalid request",
                Errors = validation.Errors.Select(e => e.ErrorMessage).ToList(),
            });
        }

        await _contactService.SubmitAsync(request, cancellationToken);
        return Ok(new ApiResponse { Success = true, Message = "Message received" });
    }
}

File: src/contact/contact.controller.ts
import { Body, Controller, Post } from '@nestjs/common';
import { ContactService } from './contact.service.js';
import { SubmitContactMessageRequestDto } from './dto/submit-contact-message-request.dto.js';
import { ApiResponse } from '../common/dto/api-response.dto.js';

@Controller('api/contact')
export class ContactController {
  constructor(private readonly contactService: ContactService) {}

  @Post()
  async submit(@Body() dto: SubmitContactMessageRequestDto) {
    await this.contactService.submit(dto);
    return ApiResponse.ok('Message received');
  }
}
